### ISSUE TITLE
(spurious?) failures in 640_proxy_restart_reattaches_test

### ISSUE THREAD

**Issue by rade** [weaveworks/weave#1588]
https://circleci.com/gh/weaveworks/weave/3387
https://circleci.com/gh/weaveworks/weave/3378
https://circleci.com/gh/weaveworks/weave/3375

**Comment by paulbellamy** [comment]
If the systemctl error is not the cause of the failure, then I would hypothesize that it is docker taking too long to reboot, so we should poll/wait, or something for that.

**Comment by bboreham** [comment]
Agreed.  This will make some test runs faster, assuming we poll quicker than we sleep atm.

**Comment by bboreham** [comment]
Per discussion in #1630 we believe there was some issue in Docker restarting containers that used `-net=host`, so I'm going to remove this from 1.2.1 milestone and we can review whether later Docker releases make this better.

**Comment by rade** [comment]
I've just run in to this and tracked down *one* of the causes:

```
INFO: 2015/11/08 10:52:55.800420 proxy listening on tcp://0.0.0.0:12375
INFO: 2015/11/08 10:52:55.803458 Attaching container 02152dcefe35aafa17a318b05a7d21a7008ddb42609afb187018f3f445d757b4 with WEAVE_CIDR "10.2.0.78/24" to weave network
WARN: 2015/11/08 10:52:55.889482 Attaching container 02152dcefe35aafa17a318b05a7d21a7008ddb42609afb187018f3f445d757b4 to weave network failed: curl: (7) Failed to connect to 127.0.0.1 port 6784: Connection refused
```
This is due to #1556 - inserting a 2 second sleep just before the call to `AttachExistingContainers` solves that particular problem.

**Comment by awh** [comment]
> I've just run in to this and tracked down one of the causes:

https://github.com/weaveworks/weave/issues/1556#issuecomment-153330840

> inserting a 2 second sleep just before the call to AttachExistingContainers 'solves' that particular problem.

Fixing this race between the proxy and the router starting up was one of the motivations for #1642.

**Comment by bboreham** [comment]
Fixed by #1561
